core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommit.java

```diff
@@ -79,40 +86,45 @@ public List<Result> scan(Scan scan) throws CrudException {
   }
 
   @Override
-  public void put(Put put) {
+  public void put(Put put) throws CrudException {
     put = copyAndSetTargetToIfNot(put);
+    checkMutation(put);
     crud.put(put);
   }
 
   @Override
-  public void put(List<Put> puts) {
+  public void put(List<Put> puts) throws CrudException {
     checkArgument(puts.size() != 0);
-    puts.forEach(this::put);
+    for (Put p : puts) {
```

Comment:
[just out of curiosity] Is this change for suppressing warning or something?

Reply:
`this::put` throws the checked exception `CrudException` but the Consumer functional interface does not throw any checked exception.
```
public void forEach(java.util.function.Consumer<? super T> action)
```
Our only option is to use a non-lambda for each if we want to propagate the `CrudException`.